### PR TITLE
fix: remove empty glob from geographiclib

### PR DESCRIPTION
**Pull request by wep21**
- as title

### CODE REVIEW

**Comment by wep21** [comment]
@bazel-io skip_check unstable_url

**Comment by wep21** [comment]
@fmeum @meteorcloudy please add presubmit auto run.

**Review by fmeum — APPROVED** [review]

File: modules/geographiclib/2.4.0.bcr.1/overlay/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+cmake_configure_file(
+    name = "config_header",
+    src = "include/GeographicLib/Config.h.in",
+    out = "include/GeographicLib/Config.h",
+    cmakelists = [
+        "CMakeLists.txt",
+    ],
+    defines = [
+        "GEOGRAPHICLIB_DATA=/usr/local/share/GeographicLib",
+        "GEOGRAPHICLIB_HAVE_LONG_DOUBLE=0",
```
Comment (kgreenek) [inline modules/geographiclib/2.4.0.bcr.1/overlay/BUILD.bazel:18]: Should this have the other variable substitutions here too? I.E.:

```
#define GEOGRAPHICLIB_VERSION_STRING "@PROJECT_VERSION@"
#define GEOGRAPHICLIB_VERSION_MAJOR @PROJECT_VERSION_MAJOR@
#define GEOGRAPHICLIB_VERSION_MINOR @PROJECT_VERSION_MINOR@
#define GEOGRAPHICLIB_VERSION_PATCH @PROJECT_VERSION_PATCH@
#define GEOGRAPHICLIB_PRECISION @GEOGRAPHICLIB_PRECISION@ 
```

Comment (wep21) [reply modules/geographiclib/2.4.0.bcr.1/overlay/BUILD.bazel:18]: `cmake_configure_file `parse `set(VAR VALUE)` from CMakeLists.txt and set it automatically.
https://github.com/wep21/cmake_configure_file/blob/ff5077867f447c59bccb7b508d0ce5f385b9f0c5/cmake_configure_file.py#L123-L143

**Comment by wep21** [comment]
@fmeum The macos ci error seems to be irrelevant to this PR for me. How do you think about it?

File: modules/geographiclib/2.4.0.bcr.1/overlay/BUILD.bazel (hunk 1 above)
Comment (kgreenek) [reply modules/geographiclib/2.4.0.bcr.1/overlay/BUILD.bazel:18]: oh cool! that's slick

**Review by kgreenek — DISMISSED** [review]

**Comment by fmeum** [comment]
The failure persists. You could check with rules_foreign_cc whether this is a known issue. Maybe there is a way to have the ruleset use a hermetic Python provided by rules_python? That would also help users of your module that don't have Python installed locally.

**Comment by kgreenek** [comment]
There are just a very small number of substitutions in this case in the Config.h file. You could consider just setting them manually in bzl and forgoing the python script that does the cmake parsing.

**Comment by wep21** [comment]
Setting default python version seems to solve the error, I'm not sure this is the proper python version though.

**Review by fmeum — APPROVED** [review]
That's okay, the root module can override it.

File: modules/geographiclib/2.4.0.bcr.1/MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    bazel_compatibility = [">=7.2.1"],
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "cmake_configure_file", version = "0.1.0")
+bazel_dep(name = "rules_cc", version = "0.1.0")
+bazel_dep(name = "rules_python", version = "1.0.0")
+python = use_extension("@rules_python//python/extensions:python.bzl", "python")
+python.toolchain(
+    is_default = True,
```
Comment (fmeum) [inline modules/geographiclib/2.4.0.bcr.1/MODULE.bazel:13]: Do tests pass without this line? That would seem safer based on https://rules-python.readthedocs.io/en/latest/toolchains.html.

File: modules/geographiclib/2.4.0.bcr.1/MODULE.bazel [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    name = "geographiclib",
+    version = "2.4.0.bcr.1",
+    bazel_compatibility = [">=7.2.1"],
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "cmake_configure_file", version = "0.1.0")
+bazel_dep(name = "rules_cc", version = "0.1.0")
+bazel_dep(name = "rules_python", version = "1.0.0")
+python = use_extension("@rules_python//python/extensions:python.bzl", "python")
```
Comment (kgreenek) [inline modules/geographiclib/2.4.0.bcr.1/MODULE.bazel:11]: Would this make more sense to be in the `cmake_configure_file` module? geographic lib doesn't directly use python anywhere.

**Review by kgreenek — DISMISSED** [review]

File: modules/geographiclib/2.4.0.bcr.1/MODULE.bazel (hunk 3 above)
Comment (wep21) [reply modules/geographiclib/2.4.0.bcr.1/MODULE.bazel]: @fmeum @kgreenek I've updated cmake configure file to use python3.12 and remove python dep in geographiclib.

**Review by fmeum — APPROVED** [review]
Great, I'll keep this helper module in mind while reviewing other contributions!

@meteorcloudy FYI

**Pull request merged**